frontend/system/src/Admin/EditDrug/helpers.ts

```diff
@@ -123,6 +124,164 @@ export const buildEntityFromDrugInput = (
   };
 };
 
+export const buildVaccineInputFromEntity = (
+  entity: EntityDetails
+): VaccineInput => {
+  return {
+    ...getDetails(entity),
+    components:
+      entity.children
+        ?.filter(component => component.type === 'component')
+        .map(component => ({
+          ...getDetails(component),
+          brands:
+            component.children
+              ?.filter(brand => brand.type === 'brand')
+              .map(brand => ({
+                ...getDetails(brand),
+                routes:
+                  brand.children
+                    ?.filter(route => route.type === 'form_category') // form_category === route
```

Comment:
// form_category === route

Since we have access to the loader code, should we just fix this in the data?
We probably need a mapping in the backend/filter to maintain backwards compatibility but might be a good plan?

Reply:
Issue for sorting this: #500